client/client.go

```diff
@@ -564,6 +564,11 @@ type (
 		SignalWithStartWorkflow(ctx context.Context, workflowID string, signalName string, signalArg interface{},
 			options StartWorkflowOptions, workflow interface{}, workflowArgs ...interface{}) (WorkflowRun, error)
 
+		// NewWithStartWorkflowOperation returns a WithStartWorkflowOperation for use with UpdateWithStartWorkflow.
+		// See [Client.UpdateWithStartWorkflow].
+		// NOTE: Experimental
+		NewWithStartWorkflowOperation(options StartWorkflowOptions, workflow interface{}, args ...interface{}) WithStartWorkflowOperation
```

Comment:
> I personally liked that this returned a validation error, but it's not that important to me

Hey! I changed it to not return an error in response to you [saying](https://github.com/temporalio/sdk-go/pull/1731#discussion_r1865922908) this:

> Can't the workflow input be created as part of UpdateWithStartWorkflow instead of on instantiation of this? One might expect the actual call to do all of the validation instead of half here and half on the actual call. Arguably this should not return an error.